clock-bound-c/src/lib.rs

```diff
@@ -339,6 +359,71 @@ impl ClockBoundClient {
             after,
         })
     }
+
+    ///Execute `f` and return bounds on execution time
+    pub fn timing<A, F>(&self, f: F) -> Result<(TimingResult, A), (ClockBoundCError, Result<A,F>)>
+        where F: FnOnce() -> A {
+
+        // Get the first timestamps
+
+        // Header
+        // 1st - Version
+        // 2nd - Command Type
+        // 3rd, 4th - Reserved
+        let mut request: [u8; 4] = [1, 1, 0, 0];
+
+        match self.socket.send(&mut request) {
+            Err(e) => return Err((ClockBoundCError::SendMessageError(e), Err(f))),
+            _ => {}
+        }
+        let mut response: [u8; 20] = [0; 20];
+        match self.socket.recv(&mut response) {
+            Err(e) => return Err((ClockBoundCError::ReceiveMessageError(e), Err(f))),
+            _ => {}
+        }
+        let response_earliest_start = NetworkEndian::read_u64(&response[4..12]);
+        let response_earliest_finish = NetworkEndian::read_u64(&response[12..20]);
+
+        // Execute the provided function, f
+        let callback = f();
+
+        // Get the second timestamps
+        let mut request: [u8; 4] = [1, 1, 0, 0];
+        match self.socket.send(&mut request) {
+            Err(e) => return Err((ClockBoundCError::SendMessageError(e), Ok(callback))),
+            _ => {}
+        }
+        let mut response: [u8; 20] = [0; 20];
+        match self.socket.recv(&mut response) {
+            Err(e) => return Err((ClockBoundCError::ReceiveMessageError(e), Ok(callback))),
+            _ => {}
+        }
+        let response_latest_start = NetworkEndian::read_u64(&response[4..12]);
+        let response_latest_finish = NetworkEndian::read_u64(&response[12..20]);
+
+        // Calculate elapsed time between early and latest timestamps
```

Comment:
You get the midpoint by *adding* and dividing by 2, not by subtracting.

Reply:
Fixed